Add Navi guide to docs (#544)
zeit/docs: pages/guides/deploying-navi-with-now.mdx

<TerminalInput>create-react-navi-app my-app</TerminalInput>
Of if you want a TypeScript project, append a --typescript flag:

<TerminalInput>create-react-navi-app my-app --typescript</TerminalInput>

After `create-react-navi` has finished generating files, navigate to your freshly project directory with

<TerminalInput>cd my-app</TerminalInput>

Finally, run your project with

<TerminalInput>yarn start</TerminalInput>

Then open http://localhost:3000/ to see your app, if it isn’t automatically opened for you.

## Step 2: Deploy Your Navi Project with Now

With your Navi set up, it is ready to deploy with [Now.](/docs/v2/getting-started/introduction-to-now)

You can prepare to deploy by creating a few pieces of configuration which will allow Now to understand how to build your project, what to deploy, and where to route users when serving your app.

The first piece is [a `now.json` configuration file](/docs/v2/deployments/configuration) file, containing the following:

```json
{
  "version": 2,
  "name": "my-app",
  "builds": [
    {
      "src": "package.json",
      "use": "@now/static-build",
      "config": { "distDir": "build" }
    }
  ]
}

> ```

Reviewer: Everything in the `/static` directory should have filenames with hashes in them, so they can be cached immutably.

Other than `/static` there are the files from `/public` which are no different from any other CRA app which probably shouldn't be cached, and HTML files which definitely should not be cached on the browser.

> ```

Author: Aside from the `/static` route should we include a route for the main `index.html` like this?
```json
  "routes": [
    {
      "src": "/static/(.*)",
      "headers": { "cache-control": "s-maxage=31536000,immutable" },
      "dest": "/static/$1"
    },
    {
      "src": "/(.*)",
      "headers": { "cache-control": "s-maxage=0" },
      "dest": "/index.html"
    }
  ]
```

> ```

Reviewer 2: Yes, I think that's best, although we can do:
```
  "routes": [
    {
      "src": "/static/(.*)",
      "headers": { "cache-control": "public,max-age=31536000,immutable" },
      "dest": "/static/$1"
    },
    {
      "src": "/(.*)",
      "headers": { "cache-control": "public,max-age=0,must-revalidate" },
      "dest": "/index.html"
    }
  ]
```

s-maxage isn't needed since Now supports both `immutable` and `must-revalidate`

> ```

Author: Ok awesome!

import Guide from '~/components/layout/guide'

import { TerminalInput } from '~/components/text/terminal'
import Card from '~/components/card'

export const meta = {

>   title: 'Deploying a React-Navi application with Now.',

Reviewer 2: Not sure it's called `React-Navi`. Perhaps the following:
```suggestion
  title: 'Deploying a Navi powered React project with Now.',
```

cc @jamesknelson

Reviewer: I haven't really got an official name for the combination of React/Navi, but your suggestion of "Deploying a Navi powered React project with Now" works well. 👍

import Guide from '~/components/layout/guide'

import { TerminalInput } from '~/components/text/terminal'
import Card from '~/components/card'

export const meta = {
  title: 'Deploying a Navi application with Now.',
  description: 'How to deploy your Navi application with Now',
  published: '1 March 2019',
  authors: ['Pr0x1m4'],
  url: '/guides/deploying-navi-with-now'
}

[Navi](https://frontarm.com/navi/en/) is modern router for React that uses Suspense, Hooks and function composition to make real-world routing simple.

In this guide, we will show you to deploy a React app that uses Navi as a router and deploy it with [ZEIT Now](/docs/v2).

## Step 1: Set Up Your Navi Project

A simple method of setting up your Navi project is by using [create-react-navi-app](https://frontarm.com/navi/en/create-react-navi-app/#create-react-navi-app). A CLI tool that
generates a project for you with all the necessary dependencies and no manual set up.

Usage is identical to the create-react-app command — you’ll just need to call create-react-navi-app instead.

> <TerminalInput>create-react-navi-app my-app</TerminalInput>

Reviewer 2: Can we explain to the user how to install this? Or even, use `yarn create react-navi-app` instead?

Reviewer 2: Also `my-navi-app` instead of `my-app` please 🙏

> Of if you want a TypeScript project, append a --typescript flag:

Reviewer 2: ```suggestion
Alternatively, if you want a TypeScript project, append the `--typescript` flag:
```

<TerminalInput>create-react-navi-app my-app --typescript</TerminalInput>

> After `create-react-navi` has finished generating files, navigate to your freshly project directory with

Reviewer 2: ```suggestion
After `create-react-navi` has finished generating files, navigate to your freshly created project directory with the following command:
```

<TerminalInput>cd my-app</TerminalInput>

Finally, run your project with

<TerminalInput>yarn start</TerminalInput>

Then open http://localhost:3000/ to see your app, if it isn’t automatically opened for you.

## Step 2: Deploy Your Navi Project with Now

> With your Navi set up, it is ready to deploy with [Now.](/docs/v2/getting-started/introduction-to-now)

Reviewer 2: ```suggestion
With your Navi project set up, it is ready to deploy with [Now.](/docs/v2/getting-started/introduction-to-now)
```

```json
{
  "version": 2,
  "name": "my-app",
  "builds": [
    {
      "src": "package.json",
      "use": "@now/static-build",
      "config": { "distDir": "build" }
    }
  ],
  "routes": [
    {
      "src": "/static/(.*)",
      "headers": { "cache-control": "public,max-age=31536000,immutable" },
      "dest": "/static/$1"
    },
    {
      "src": "/(.*)",
      "headers": { "cache-control": "public,max-age=0,must-revalidate" },
      "dest": "/index.html"
    }
  ]
}
```

To explain the above briefly, this now.json file tells Now to:

- Build using [the latest Now 2.0 platform](/docs/v2/platform/overview/#versioning).
- Set a [project name](/docs/v2/deployments/configuration/#name) for all deployments.
- Build your application using the [@now/static-build](/docs/v2/deployments/official-builders/static-build-now-static-build/) builder and configure the builder to look in the build directory for the built app.

By default, create-react-navi-app will provide build scripts, and those scripts will output the `build` directory. The only change you need to make is to create a `now-build` script in the `package.json`:

```json
{
  "scripts": {
    ...

>     "now-build": "npm run build"

Reviewer 2: Can this use the build script value directly instead? Just to be very clear.

  }
}
```

> Then, you can deploy from your terminal with just one command:

Reviewer 2: ```suggestion
Then, you can deploy your project from your terminal with just one command:
```
